core/env/clone_spec.rb

```diff
@@ -0,0 +1,29 @@
+require_relative '../../spec_helper'
+
+describe "ENV#clone" do
+  ruby_version_is "3.2" do
+    it "raises TypError" do
+      message = /Cannot clone ENV, use ENV.to_h to get a copy of ENV as a hash/
+
+      -> {
+        ENV.clone(freeze: nil)
+      }.should raise_error(TypeError, message)
+
+      -> {
+        ENV.clone
+      }.should raise_error(TypeError, message)
+    end
+
+    it "raises ArgumentError when keyword argument 'freeze' is neither nil nor boolean" do
+      -> {
+        ENV.clone(freeze: 1)
+      }.should raise_error(ArgumentError)
+    end
+
+    it "raises ArgumentError when keyword argument is not 'freeze'" do
+      -> {
+        ENV.clone(foo: nil)
+      }.should raise_error(ArgumentError)
+    end
```

Comment:
**minor:** I suppose `ArgumentError` cases aren't Ruby 3.2 specific and could be moved out of this version guard.